### PR TITLE
feat: Fix icons used in menu items for dropdown menu's

### LINKED ISSUE #925: [Improvement request]: Fix icons used in menu items for dropdown menu's
### Contact Details

jurre@shelf.nu

### Is your feature request related to a problem? Please describe?

We're not consistently using a container for the icons used in menu items which results in the icons looking off. It's not something that is very high impact but it doesn't look very professional.

EXAMPLE:

Screenshot 2024-04-23 at 17 38 54
Screenshot 2024-04-23 at 17 41 56

### Describe the solution you'd like

Identify all places within the app where these dropdown menus are used and giving the icons a consistent container. Maybe there is way to create a global component for it so from now on it can stay consistent.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Bhavyajain21**
/fixes #925 
/claim #925 

image

### CODE REVIEW

**Review by DonKoko — CHANGES_REQUESTED** [review]
Some changes required based on our discussion on discord.

File: app/components/assets/actions-dropdown.tsx [hunk 1]
```diff
@@ -86,7 +87,8 @@ const ConditionalActionsDropdown = () => {
                 [data-radix-popper-content-wrapper] {
                   transform: none !important;
                   will-change: auto !important;
-              }
+              }import iconsMap from '../shared/icons-map';
```
Comment (DonKoko) [inline app/components/assets/actions-dropdown.tsx]: Not sure what happened here. This shouldn't be there.

File: app/components/icons/WrapIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import IconHug from "./IconHug";
+import type { Icon } from "../shared/icons-map";
+import iconsMap from "../shared/icons-map";
+
+export interface WrapIconProps {
+  icon?: Icon;
+  enableWrap?: boolean;
+}
+const WrapIcon = React.forwardRef<HTMLElement, WrapIconProps>(
```
Comment (DonKoko) [inline app/components/icons/WrapIcon.tsx]: Rename the component to simply `Icon`

File: app/components/icons/WrapIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  function WrapIcon({ icon, enableWrap }: WrapIconProps) {
```
Comment (DonKoko) [inline app/components/icons/WrapIcon.tsx]: `enableWrap` should be `true` by default so we dont have to pass it every time, only when we want it to be false as that will be the more rare case.

File: app/components/assets/actions-dropdown.tsx (hunk 1 above)
Comment (Bhavyajain21) [reply app/components/assets/actions-dropdown.tsx]: Resolved, I think it got added by mistake!

File: app/components/icons/WrapIcon.tsx (hunk 2 above)
Comment (Bhavyajain21) [reply app/components/icons/WrapIcon.tsx]: Replaced Icon as IconType, as we were seeing some errors while import the component with the same name as that of Type

File: app/components/icons/WrapIcon.tsx (hunk 3 above)
Comment (Bhavyajain21) [reply app/components/icons/WrapIcon.tsx]: Added a disableWrap prop, which needs to be passed when wrapping is not required

**Comment by Bhavyajain21** [comment]
Made the changes accordingly!

**Review by DonKoko — APPROVED** [review]

**Pull request merged**